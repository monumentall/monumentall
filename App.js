import React from "react";
import { Platform, StatusBar, StyleSheet, View } from "react-native";
import { Provider } from "react-redux";
import store from "./store";
import { AppLoading } from "expo";
import * as Font from "expo-font";
import HomeScreen from "./screens/HomeScreen";

export default class App extends React.Component {
  state = {
    isLoadingComplete: false
  };

  render() {
    if (!this.state.isLoadingComplete && !this.props.skipLoadingScreen) {
      return (
        <AppLoading
          startAsync={this._loadResourcesAsync}
          onError={this._handleLoadingError}
          onFinish={this._handleFinishLoading}
        />
      );
    } else {
      return (
        <Provider store={store}>
          <View style={styles.container}>
            {Platform.OS === "ios" && <StatusBar barStyle="default" />}
            <HomeScreen />
          </View>
        </Provider>
      );
    }
  }

  _loadResourcesAsync = async () => {
    return Promise.all([
      Font.loadAsync({
        nunito: require("./assets/fonts/Nunito/Nunito-Regular.ttf"),
        "nunito-bold": require("./assets/fonts/Nunito/Nunito-Bold.ttf")
      })
    ]);
  };

  _handleLoadingError = error => {
    console.warn(error);
  };

  _handleFinishLoading = () => {
    this.setState({ isLoadingComplete: true });
  };
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff"
  }
});
