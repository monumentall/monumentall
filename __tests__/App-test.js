import React from "react";
import { configure, shallow } from "enzyme";
import App from "../App";
import Adapter from "enzyme-adapter-react-16";

configure({
  adapter: new Adapter()
});

const baseProps = {
  skipLoadingScreen: true
};

describe("App", () => {
  it(`renders the loading app if skipLoadingScreen is false`, () => {
    const props = { ...baseProps, skipLoadingScreen: false };

    const wrapper = shallow(<App {...props} />);
    expect(wrapper.find("AppLoading")).toHaveLength(1);
  });

  it(`renders the homeScreen if skipLoadingScreen is true`, () => {
    const props = { ...baseProps };

    const wrapper = shallow(<App {...props} />);
    expect(wrapper.find("Provider")).toHaveLength(1);
  });
});
