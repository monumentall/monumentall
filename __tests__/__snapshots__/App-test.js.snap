// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the loading screen 1`] = `
<AppLoading
  onError={[Function]}
  onFinish={[Function]}
  startAsync={[Function]}
/>
`;

exports[`App renders the root without loading screen 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
      }
    }
  >
    <HomeScreen />
  </View>
</Provider>
`;
